### ISSUE TITLE
The suite stopped working with the most recent Kali version (2019.2 with all updates)

### ISSUE THREAD

**Issue by pentestdude** [vanhoefm/krackattacks-scripts#71]
Hi,

I have been using the suite in various pentests with success (not so much for various vendors...). After a recent Kali update, the suite suddenly stopped working. In order to confirm that issue, I've set up a fresh Kali installation in a virtual machine with the most recent updates. I got the same result:

Running client test 1 (replay broadcast) will always result in "Reset PN for GTK" being printed forever (as seen in the screenshot below). At the same time the suite works fine on an old Kali version (2019.1) lacking any recent updates running on bare metal.
For all three "machines" I used the same Wi-Fi adapter (Alfa Network AWUS036NHA).

@vanhoefm would it be possible to check the suite with the most recent version of Kali including all updates and verify that it is possibly broken? 
Could anyone else confirm the issue (and possibly try different Wi-Fi adapter, maybe this is a driver issue)?

![grafik](https://user-images.githubusercontent.com/54318757/63352494-80b37780-c361-11e9-8564-dd4c5d2b1acb.png)

Thanks a lot for all your work on this!

**Comment by JMan3333** [comment]
I am using Kali 2019.2 and I've faced this problem before. I might need some time to confirm and check whether I set up wrongly that brought up this problem

**Comment by vanhoefm** [comment]
Did you try another Wi-Fi dongle as well?

In case more people face this issue, please let me know.

**Comment by JMan3333** [comment]
![Alfa 1](https://user-images.githubusercontent.com/54394303/63503936-0a4e7b00-c503-11e9-85b2-0444f713966e.PNG)
I got this result by using Alfa Atheros AR9271, using Kali Linux 2019.02 I've also followed the instruction provided to set up the script and edit the hostapd.conf file.

Edit: I've also used Alfa Network AWUS036NHR and got the same results

**Comment by pentestdude** [comment]
@vanhoefm, no I only used the Alfa Adapter since it has been very reliable in the past. I will try to get another adapter to run another test.

**Comment by pentestdude** [comment]
@JMan3333, yes that's how it looks for me, I just didn't add the full screenshot. The important part is the "Reset PN for GTK" at the end of your output which will go on like that forever without actually starting the test properly.

**Comment by JMan3333** [comment]
@pentestdude Thanks for verifying for me, I thought that some of my settings I've set may be wrong. I've also tried on an older version of Kali Linux which is 2019.01. But I still got the same results, the only problem would be the network Adapter @vanhoefm mentioned. I will try and see if changing the network adapter will work.

**Comment by pentestdude** [comment]
A colleague of mine tried different adapters today. The problem is indeed with the Alfa Network adapter. We tried the following ALFA Adapter which **don't work any longer**:

- ALFA AWUS036ACH (Realtek RTL8812AU) with kernel module 88XXau
- ALFA AWUS036NHA (Atheros AR9271) with kernel module ath9k_htc

The following adapter **worked unreliable** and should propably not be used:

- Edimax EW-7612UAn V2 (Realtek RTL8191SU) with kernel module rtl8192cu

In addition we tried the following adapters which **worked**:

- Hama Wireless Lan (Ralink RT2501/RT2573) - with kernel module rt73usb
- Intel Wireless 7265 - with kernel module iwlwifi

So the problem is **with some adapters and drivers**. It is interesting that the ALFA both don't work, though they have different hardware and driver.

This is how it looked with the Hama Wireless Lan (Ralink RT2501/RT2573):
![grafik](https://user-images.githubusercontent.com/54318757/63771201-f4600200-c8d6-11e9-8534-b0ec3cd06a63.png)

**Comment by pentestdude** [comment]
It would be nice to have a list of working adapters. @vanhoefm would it be possible to add such a list?

**Comment by gabba90** [comment]
Hi,

I am using Kali 2019.3 and I also faced this issue. I found a solution and now it seems to work again. 
Briefly, I replaced the pre-installed dependency 'python-scapy 2.4.2' with a previous one, that is 'python-scapy 2.4.0'.

These are the steps:
1. Install dependencies:
	
       apt-get update
       apt-get install libnl-3-dev libnl-genl-3-dev pkg-config libssl-dev net-tools git sysfsutils python-pycryptodome

2. Uninstall the pre-installed 'python-scapy':
	
       apt-get purge python-scapy
       apt autoremove

3. Install 'python-scapy 2.4.0':
	
       apt-get install python-pip
       pip install scapy==2.4.0

4. Clone 'krackattacks-scripts':
	
       git clone https://github.com/vanhoefm/krackattacks-scripts.git

5. Disable hardware encryption and reboot:
	
       ./krackattacks-scripts/krackattack/disable-hwcrypto.sh
       reboot

6. Compile modified hostapd instance:
	
       cd hostapd
       cp defconfig .config
       make -j 2

7. Disable Wi-Fi

8. Soft-Unblock Wi-Fi:

       rfkill unblock wifi

9. Modify 'hostapd/hostapd.conf' as explained in the README file

10. Finally, execute the scripts. For example:

        ./krackattacks-scripts/krackattack/krack-test-client.py

	
I tested this procedure with: 

- TP-Link TL-WN722N v1 (Atheros AR9271);
- TP-Link TL-WN823N (Realtek RTL8192CU);
- Raspberry Pi 3 Wi-Fi module (Broadcom bcm43430).

Thanks @vanhoefm for scripts!

**Comment by zanaries** [comment]
> Hi,
> 
> I am using Kali 2019.3 and I also faced this issue. I found a solution and now it seems to work again.
> Briefly, I replaced the pre-installed dependency 'python-scapy 2.4.2' with a previous one, that is 'python-scapy 2.4.0'.
> 
> These are the steps:
> 
>     1. Install dependencies:
>        ```
>        apt-get update
>        apt-get install libnl-3-dev libnl-genl-3-dev pkg-config libssl-dev net-tools git sysfsutils python-pycryptodome
>        ```
> 
>     2. Uninstall the pre-installed 'python-scapy':
>        ```
>        apt-get purge python-scapy
>        apt autoremove
>        ```
> 
>     3. Install 'python-scapy 2.4.0':
>        ```
>        apt-get install python-pip
>        pip install scapy==2.4.0
>        ```
> 
>     4. Clone 'krackattacks-scripts':
>        ```
>        git clone https://github.com/vanhoefm/krackattacks-scripts.git
>        ```
> 
>     5. Disable hardware encryption and reboot:
>        ```
>        ./krackattacks-scripts/krackattack/disable-hwcrypto.sh
>        reboot
>        ```
> 
>     6. Compile modified hostapd instance:
>        ```
>        cd hostapd
>        cp defconfig .config
>        make -j 2
>        ```
> 
>     7. Disable Wi-Fi
> 
>     8. Soft-Unblock Wi-Fi:
>        ```
>        rfkill unblock wifi
>        ```
> 
>     9. Modify 'hostapd/hostapd.conf' as explained in the README file
> 
>     10. Finally, execute the scripts. For example:
>         ```
>         ./krackattacks-scripts/krackattack/krack-test-client.py
>         ```
> 
> 
> I tested this procedure with:
> 
>     * TP-Link TL-WN722N v1 (Atheros AR9271);
> 
>     * TP-Link TL-WN823N (Realtek RTL8192CU);
> 
>     * Raspberry Pi 3 Wi-Fi module (Broadcom bcm43430).
> 
> 
> Thanks @vanhoefm for scripts!

Where is the README file to modify hostapd.conf file?

**Comment by gabba90** [comment]
> > Hi,
> > I am using Kali 2019.3 and I also faced this issue. I found a solution and now it seems to work again.
> > Briefly, I replaced the pre-installed dependency 'python-scapy 2.4.2' with a previous one, that is 'python-scapy 2.4.0'.
> > These are the steps:
> > ```
> > 1. Install dependencies:
> >    ```
> >    apt-get update
> >    apt-get install libnl-3-dev libnl-genl-3-dev pkg-config libssl-dev net-tools git sysfsutils python-pycryptodome
> >    ```
> > 
> > 2. Uninstall the pre-installed 'python-scapy':
> >    ```
> >    apt-get purge python-scapy
> >    apt autoremove
> >    ```
> > 
> > 3. Install 'python-scapy 2.4.0':
> >    ```
> >    apt-get install python-pip
> >    pip install scapy==2.4.0
> >    ```
> > 
> > 4. Clone 'krackattacks-scripts':
> >    ```
> >    git clone https://github.com/vanhoefm/krackattacks-scripts.git
> >    ```
> > 
> > 5. Disable hardware encryption and reboot:
> >    ```
> >    ./krackattacks-scripts/krackattack/disable-hwcrypto.sh
> >    reboot
> >    ```
> > 
> > 6. Compile modified hostapd instance:
> >    ```
> >    cd hostapd
> >    cp defconfig .config
> >    make -j 2
> >    ```
> > 
> > 7. Disable Wi-Fi
> > 
> > 8. Soft-Unblock Wi-Fi:
> >    ```
> >    rfkill unblock wifi
> >    ```
> > 
> > 9. Modify 'hostapd/hostapd.conf' as explained in the README file
> > 
> > 10. Finally, execute the scripts. For example:
> >     ```
> >     ./krackattacks-scripts/krackattack/krack-test-client.py
> >     ```
> > ```
> > 
> > 
> > I tested this procedure with:
> > ```
> > * TP-Link TL-WN722N v1 (Atheros AR9271);
> > 
> > * TP-Link TL-WN823N (Realtek RTL8192CU);
> > 
> > * Raspberry Pi 3 Wi-Fi module (Broadcom bcm43430).
> > ```
> > 
> > 
> > Thanks @vanhoefm for scripts!
> 
> Where is the README file to modify hostapd.conf file?

https://github.com/vanhoefm/krackattacks-scripts/blob/research/README.md

**Comment by a-posadas** [comment]
I just logged in to say I was facing the same issue. I tried with two different wifi cards and it didn't seem to make a difference. However, the scapy downgrade seems to resolve the issue. Thanks @Gabba90

**Comment by vanhoefm** [comment]
The code has been updated to be compatible with more recent scapy versions. If there are still problems, you can report them.
